index.d.ts

```diff
@@ -16,10 +16,14 @@ interface EggRedisOptions {
 
 declare module 'egg' {
   interface Application {
-    redis: Redis;
+    redis: Redis & Singleton<Redis>;
   }
 
   interface EggAppConfig {
     redis: EggRedisOptions;
   }
+
+  class Singleton<T> {
```

Comment:
`Singleton` support has published a new version( https://github.com/eggjs/egg/releases/tag/2.20.1 )

So you can remove this declaration.

Reply:
ok